### PR TITLE
added meal and reservation end points

### PR DESCRIPTION
**Pull request by shobanamg**

### CODE REVIEW

File: src/backend/controllers/mealController.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        console.error(err);
+        res.status(500).json({message: 'Failed to retrieve the meal'});
+    }
+}
+
+const updateMealById = async (req, res) => {
+    try {
+        const result = await Meal.updateMealById(req.params.id, req.body);
+        result ?
+            res.status(200).json('Meal updated') :
```
Comment (YurijPaharev) [inline src/backend/controllers/mealController.js]: If you are sending json as a response, it is better to be consistent and send it in a structured way like `{message: 'Meal updated'}`, or `{error: 'No meal found'}`

File: src/backend/models/meal.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        .orderBy('id' , 'desc')
+        .first()
+        .catch(err => {
+            console.error(err);
+            throw err;
+        });
+}
+
+const getMealById = async (id) => {
+    return knex
```
Comment (YurijPaharev) [inline src/backend/models/meal.js]: Please check if all async/await pairs are provided correctly

**Review by YurijPaharev — COMMENTED** [review]
Please check some comments

File: src/backend/controllers/mealController.js (hunk 1 above)
Comment (shobanamg) [reply src/backend/controllers/mealController.js]: It makes sense, i have updated them, in all the places

File: src/backend/models/meal.js (hunk 2 above)
Comment (shobanamg) [reply src/backend/models/meal.js]: I have missed it in few places, fixed all of them.

**Comment by shobanamg** [comment]
> Please check some comments
@YurijPaharev, Thanks again for the wonderful comments. Fixed all the of review comments

**Review by YurijPaharev — APPROVED** [review]

**Pull request merged**